projector/binding.go

```diff
@@ -302,7 +299,7 @@ func (p *serviceBindingProjector) unprojectEnv(binding *servicebindingv1alpha3.S
 }
 
 func (p *serviceBindingProjector) isContainerBindable(binding *servicebindingv1alpha3.ServiceBinding, mc *metaContainer) bool {
-	if len(binding.Spec.Workload.Containers) == 0 || mc.Name == "" {
+	if len(binding.Spec.Workload.Containers) == 0 || !mc.NameWasMapped() {
```

Comment:
why do we need here `!mc.NameWasMapped()`? Should not be enough that `len(binding.Spec.Workload.Containers) == 0`? That means that the binding did not specify any container names so, all containers present in the workload are bindable, right?

Reply:
`len(binding.Spec.Workload.Containers) == 0` means that the `ServiceBinding` didn't restrict binding to a set of named containers. It has no bearing on whether any containers where mapped (or container names for that matter).

The `NameWasMapped()` method indicates that the the name field was mapped from the corev1.Container-like object to the metaContainer object. This is important because we need to distinguish between the mapping not defining a path for the container name, and the container-like struct not specifying a name (or an empty name). The other option would be to switch the metaContainer name field's type to `*string` and distinguishing nil from empty.

Reply:
> If either key is not defined, the container SHOULD be bound.

Thanks, I missed that.

> The other option would be to switch the metaContainer name field's type to *string and distinguishing nil from empty.

+1 for that - keeping all semantic in the single field.

Reply:
updated the type to `*string` and removed the NameWasMapped method